lib/CMakeLists.txt

```diff
@@ -100,6 +90,24 @@ set (QUDA_OBJS
   instantiate.cpp version.cpp )
 # cmake-format: on
 
+# current workaround for nvshmem
+set(QUDA_DSLASH_OBJS 
+  dslash_staggered.cu dslash_improved_staggered.cu
+  dslash_wilson.cu dslash_wilson_clover.cu dslash5_domain_wall.cu
+  dslash_wilson_clover_preconditioned.cu 
+  dslash_twisted_mass.cu dslash_twisted_mass_preconditioned.cu
+  dslash_ndeg_twisted_mass.cu dslash_ndeg_twisted_mass_preconditioned.cu
+  dslash_twisted_clover.cu dslash_twisted_clover_preconditioned.cu
+  dslash_wilson_clover_hasenbusch_twist.cu
+  dslash_wilson_clover_hasenbusch_twist_preconditioned.cu
+  dslash_domain_wall_4d.cu  dslash_domain_wall_5d.cu
+  mdw_fused_dslash.cu dslash5_mobius_eofa.cu
```

Comment:
I don't think there's any need for mdw_fused_dslash.cu and dslash5_mobius_eofa.cu to be listed here.  There are no communications in these kernels.

Reply:
Looking through that it seems that `mdw_fused_dslash.cu` triggers some warnings (nvshmem extern being treated as static). I wonder if cleaning up the `instantiate` there can help. @hummingtree Do you think it is possible to move the `instantiatePreconditioner` out of `dslash.h` into `instantiate` and make it act more like the other instantiate things there?

I think if we can remove the include `dslash.h` and only include what is actually need we might be able to clean that up.

Reply:
If I change the includes for `mdw_fused_dslash.cu` like

```
-#include <dslash.h>
+
+#include <typeinfo>
+
+#include <color_spinor_field.h>
+#include <tune_quda.h>
+#include <dslash_quda.h>
+#include <jitify_helper.cuh>
+#include <instantiate.h>
```

and add all the `instantiate` from dslash.h explicitly I can pull it out of the `dslash_pack` target. 

Do we want to move them into a new header? Does this conflict with GK somehow? In that case we might want to fix that later and leave it as it is for now.

Reply:
I will try cleaning this up.

Reply:
The critical thing is that it should not include `dslash_helper.cuh` in any way since this pull in nvshmem headers which trigger the warning when compiled without `rdc=true` (which is the case for the dslash*.cu files in the list above).